### DISCUSSION TITLE
pymdown's Tabbed extension

### DISCUSSION THREAD

**Discussion by starpit** [rehypejs/rehype#81] [Ideas] [answered]
pymdown has a nice [Tabbed extension](https://facelessuser.github.io/pymdown-extensions/extensions/tabbed)

I thought I would take a quick stab at a rehype plugin for this. It seems fairly straightforward, with one exception: pymdown does something strange (in my opinion) w.r.t. indentation. The content of each tab is indented, which runs contrary to the normal interpretation of indentation in markdown.

Am I correct in guessing that doing this at the rehype level would not be possible (at least, coming at this from `react-markdown`)? As far as I can tell, my rehype extension is given AST nodes that already have `<pre>` for indented blocks... so a bit too late of an intercept point?

Here you can see that I kinda got it working, but without the indentation that pymdown supports (requires?)

## Screenshot of my hack

<img width="837" alt="Screen Shot 2021-12-08 at 2 34 40 PM" src="https://user-images.githubusercontent.com/4741620/145272476-a1a95add-c39d-4bcc-ae9a-1dbcb561eb3a.png">

**Comment by wooorm** [comment, marked as answer]
If you’re going to go with non-standard extensions to the language, I recommend using [directives](https://github.com/remarkjs/remark-directive) as it’s at least well thought out and solves many use cases, and it’s the most likely to become standard.
More info on whether it’s a good idea to extend markdown and how to do it at different levels is detailed here: https://github.com/micromark/micromark#extensions.
